File: src/data/scripts/campaign/plugins/niko_MPC_satelliteInteractionDialogPlugin.java
package data.scripts.campaign.plugins;

import com.fs.starfarer.api.campaign.BattleAPI;
import com.fs.starfarer.api.campaign.CampaignFleetAPI;
import com.fs.starfarer.api.impl.campaign.FleetInteractionDialogPluginImpl;
import data.utilities.niko_MPC_fleetUtils;

import java.util.List;

public class niko_MPC_satelliteInteractionDialogPlugin extends FleetInteractionDialogPluginImpl {

    @Override
    protected boolean fleetWantsToDisengage(CampaignFleetAPI fleet, CampaignFleetAPI other) {
        boolean result = super.fleetWantsToDisengage(fleet, other);
        BattleAPI battle = context.getBattle();

        if (battle != null) {
            List<CampaignFleetAPI> fleetsForFirst = battle.getSideFor(fleet);
            for (CampaignFleetAPI potentialSatellite : fleetsForFirst) {
                if (niko_MPC_fleetUtils.fleetIsSatelliteFleet(potentialSatellite)) {
                    return false;
                }
            }
        }
        return result;
    }

    @Override
    protected boolean fleetWantsToFight(CampaignFleetAPI fleet, CampaignFleetAPI other) {
        boolean result = super.fleetWantsToFight(fleet, other);

        BattleAPI battle = context.getBattle();

        if (battle != null) {
            List<CampaignFleetAPI> fleetsForFirst = battle.getSideFor(fleet);
            for (CampaignFleetAPI potentialSatellite : fleetsForFirst) {
                if (niko_MPC_fleetUtils.fleetIsSatelliteFleet(potentialSatellite)) {
                    result = super.fleetWantsToFight(potentialSatellite, other);
                    break;
                }
            }
        }
        return result;
    }

    @Override
    protected boolean fleetHoldingVsStrongerEnemy(CampaignFleetAPI fleet, CampaignFleetAPI other) {
        boolean result = super.fleetHoldingVsStrongerEnemy(fleet, other);

        BattleAPI battle = context.getBattle();

        if (battle != null) {
            List<CampaignFleetAPI> fleetsForFirst = battle.getSideFor(fleet);
            for (CampaignFleetAPI potentialSatellite : fleetsForFirst) {
                if (niko_MPC_fleetUtils.fleetIsSatelliteFleet(potentialSatellite)) {
                    result = super.fleetHoldingVsStrongerEnemy(potentialSatellite, other);
                    break;
                }
            }
        }
        return result;
    }
}

File: src/data/scripts/campaign/plugins/niko_MPC_satelliteInteractionDialogPlugin.kt
package data.scripts.campaign.plugins

import com.fs.starfarer.api.campaign.CampaignFleetAPI
import com.fs.starfarer.api.impl.campaign.FleetInteractionDialogPluginImpl
import data.utilities.niko_MPC_fleetUtils.isSatelliteFleet

//todo: consider removing, now that i have it so that satellites flip to the other side of the map on pursuits
class niko_MPC_satelliteInteractionDialogPlugin : FleetInteractionDialogPluginImpl() {
    override fun fleetWantsToDisengage(fleet: CampaignFleetAPI, other: CampaignFleetAPI): Boolean {
        val result = super.fleetWantsToDisengage(fleet, other)
        val battle = context.battle
        if (battle != null) {
            val fleetsForFirst = battle.getSideFor(fleet)
            for (potentialSatellite in fleetsForFirst) {
                if (potentialSatellite.isSatelliteFleet()) {
                    return false
                }
            }
        }
        return result
    }

    override fun fleetWantsToFight(fleet: CampaignFleetAPI, other: CampaignFleetAPI): Boolean {
        var result = super.fleetWantsToFight(fleet, other)
        val battle = context.battle
        if (battle != null) {
            val fleetsForFirst = battle.getSideFor(fleet)
            for (potentialSatellite in fleetsForFirst) {
                if (potentialSatellite.isSatelliteFleet()) {
                    result = super.fleetWantsToFight(potentialSatellite, other)
                    break
                }
            }
        }
        return result
    }

    override fun fleetHoldingVsStrongerEnemy(fleet: CampaignFleetAPI, other: CampaignFleetAPI): Boolean {
        var result = super.fleetHoldingVsStrongerEnemy(fleet, other)
        val battle = context.battle
        if (battle != null) {
            val fleetsForFirst = battle.getSideFor(fleet)
            for (potentialSatellite in fleetsForFirst) {
                if (potentialSatellite.isSatelliteFleet()) {
                    result = super.fleetHoldingVsStrongerEnemy(potentialSatellite, other)
                    break
                }
            }
        }
        return result
    }
}